synapse/storage/schema/main/delta/59/11add_knock_members_to_stats.sql

```diff
@@ -0,0 +1,16 @@
+/* Copyright 2020 Sorunome
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ALTER TABLE room_stats_current ADD COLUMN knocked_members INT NOT NULL DEFAULT '0';
```

Comment:
It turns out we do actually use the `room_stats_historical` table:

https://github.com/matrix-org/synapse/blob/44ce919c706fba46899dc496ab0ef244ac648bd7/synapse/storage/databases/main/stats.py#L330

As do the tests:

https://github.com/matrix-org/synapse/blob/44ce919c706fba46899dc496ab0ef244ac648bd7/synapse/storage/databases/main/stats.py#L330

Whether the data in there is actually useful is debatable, but at least for the purposes of this PR it's necessary to pass the tests.

Reply:
That's fair enough then! 👍